### PR TITLE
Jenkins: separate build pipeline and package test pipeline

### PR DESCRIPTION
**Pull request by micahsnyder**
The build pipeline used to build clamav packages and then test those packages with a rudimentary test set. This change will build the clamav packages in one pipeline - then test the packages in a new test pipeline.

The new test pipeline will use the larger test suite that we use for testing from-source builds in the "regular" test pipeline.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -82,53 +108,78 @@ node('ubuntu-18-x64') {
     def buildResult
 
     stage('Build') {
-        buildResult = build(job: "${params.TEST_PIPELINES_PATH}/${params.BUILD_PIPELINE}",
+        buildResult = build(job: "${params.BUILD_PIPELINES_PATH}/${params.BUILD_PIPELINE}",
             propagate: true,
             wait: true,
             parameters: [
+                [$class: 'StringParameterValue', name: 'PIPELINE_BRANCH_NAME', value: "${params.BUILD_PIPELINE}"],
```
Comment (rsundriyal) [inline Jenkinsfile:115]: Not required anymore. We can remove it.

File: Jenkinsfile [hunk 2]
```diff
@@ -82,53 +108,78 @@ node('ubuntu-18-x64') {
...
             def exception = null
             try {
-                stage("Regular Pipeline") {
-                    regularResult = build(job: "${params.TEST_PIPELINES_PATH}/${params.REGULAR_PIPELINE}",
+                stage("Package") {
+                    final regularResult = build(job: "${params.TEST_PIPELINES_PATH}/${params.PACKAGE_PIPELINE}",
                         propagate: true,
                         wait: true,
                         parameters: [
+                            [$class: 'StringParameterValue', name: 'PIPELINE_BRANCH_NAME', value: "${params.PACKAGE_PIPELINE}"],
```
Comment (rsundriyal) [inline Jenkinsfile:137]: Need to be removed.

File: Jenkinsfile [hunk 3]
```diff
@@ -82,53 +108,78 @@ node('ubuntu-18-x64') {
...
+                exception = exc
+            }
+
+            try {
+                stage("Regular From-Source") {
+                    final regularResult = build(job: "${params.TEST_PIPELINES_PATH}/${params.REGULAR_PIPELINE}",
+                        propagate: true,
+                        wait: true,
+                        parameters: [
+                            [$class: 'StringParameterValue', name: 'PIPELINE_BRANCH_NAME', value: "${params.REGULAR_PIPELINE}"],
```
Comment (rsundriyal) [inline Jenkinsfile:161]: Need to be removed.

File: Jenkinsfile [hunk 4]
```diff
@@ -82,53 +108,78 @@ node('ubuntu-18-x64') {
...
                 exception = exc
             }
-            stage("Custom Pipeline") {
+
+            stage("Custom From-Source") {
                 final customResult = build(job: "${params.TEST_PIPELINES_PATH}/${params.CUSTOM_PIPELINE}",
                     propagate: true,
                     wait: true,
                     parameters: [
+                        [$class: 'StringParameterValue', name: 'PIPELINE_BRANCH_NAME', value: "${params.CUSTOM_PIPELINE}"],
```
Comment (rsundriyal) [inline Jenkinsfile:182]: Need to be removed. Not required.

**Review by rsundriyal — COMMENTED** [review]
Minor changes.

File: Jenkinsfile (hunk 1 above)
Comment (micahsnyder) [reply Jenkinsfile]: Done

File: Jenkinsfile (hunk 3 above)
Comment (micahsnyder) [reply Jenkinsfile]: Done

File: Jenkinsfile (hunk 2 above)
Comment (micahsnyder) [reply Jenkinsfile]: Done

File: Jenkinsfile (hunk 4 above)
Comment (micahsnyder) [reply Jenkinsfile]: Done

**Review by rsundriyal — APPROVED** [review]
Looks great.

**Pull request merged**